packages/typescript/internal/build_defs.bzl

```diff
@@ -155,12 +155,17 @@ def _compile_action(ctx, inputs, outputs, tsconfig_file, node_opts, description
         # rather than the contents getting expanded.
         arguments.append("@@" + tsconfig_file.path)
 
+        unused_inputs_list = ctx.actions.declare_file("{}.{}.unused".format(ctx.label.name, description))
+        arguments.append("@@" + unused_inputs_list.path)
+        action_outputs.append(unused_inputs_list)
+
         # Spawn a plain action that runs worker process with no linker
         ctx.actions.run(
             progress_message = "Compiling TypeScript (%s) %s" % (description, ctx.label),
             mnemonic = "TypeScriptCompile",
             inputs = action_inputs,
             outputs = action_outputs,
+            unused_inputs_list = unused_inputs_list,
```

Comment:
do we also need to plumb the unused inputs list in the non-worker case below?

Reply:
It needs the list of all inputs in order to derive which are unused.  Workers have that by default so I only enabled it for them.
The non-worker code behaves sanely, but just can't take advantage of it.
We could pass in the names of all of the inputs through the tsconfig, but that can be added later, so I decided to not go gallivanting around for it.